lib/plugin/videoroom.ex

```diff
@@ -31,38 +38,305 @@ defmodule Janus.Plugin.VideoRoom do
     :require_e2ee
   ]
 
-  def create_room(room_name, room_properties, connection, session_id, handle_id) do
-    message = new_room_message(room_name, room_properties, session_id, handle_id)
+  @type audio_codec :: :opus | :g722 | :pcmu | :pcma | :isac32 | :isac16
+  @type video_codec :: :vp8 | :vp9 | :h264 | :av1 | :h265
+  @type t :: %__MODULE__{
+          description: binary() | nil,
+          is_private: boolean() | nil,
+          secret: binary() | nil,
+          pin: binary() | nil,
+          require_pvtid: boolean() | nil,
+          publishers: non_neg_integer() | nil,
+          bitrate: non_neg_integer() | nil,
+          bitrate_cap: boolean() | nil,
+          fir_freq: non_neg_integer() | nil,
+          audiocodec: list(audio_codec) | nil,
+          videocodec: list(video_codec) | nil,
+          vp9_profile: binary() | nil,
+          h264_profile: binary() | nil,
+          opus_fec: boolean() | nil,
+          video_svc: boolean() | nil,
+          audiolevel_ext: boolean() | nil,
+          audiolevel_event: boolean() | nil,
+          audio_active_packets: non_neg_integer() | nil,
+          # Float between 0 and 127
+          # 127 means complete silence
+          # 0 loud
+          # default is 25
+          audio_level_average: float() | nil,
+          videoorient_ext: boolean() | nil,
+          playoutdelay_ext: boolean() | nil,
+          transport_wide_cc_ext: boolean() | nil,
+          record: boolean() | nil,
+          rec_dir: boolean() | nil,
+          lock_record: boolean() | nil,
+          notify_joining: boolean() | nil,
+          require_e2ee: boolean() | nil
+        }
+
+  @type room_id :: String.t()
+  @type room :: map
+  @type room_properties :: struct
+  @type handle_id :: String.t()
+  @type admin_key :: String.t() | nil
+  @type room_secret :: String.t() | nil
+  @type action :: String.t()
+  @type allowed :: list(String.t())
+
+  @type participant :: map
+
+  @doc """
+  Sends request to create a new room.
+
+  ## Arguments
+  * `session` - valid `Janus.Session` process to send request through
+  * `room_id` - new room's id, if set to nil gateway will create one
+  * `room_properties` - map containing necessary information about room
```

Comment:
You can put here type reference to make list of viable options more visible.

Reply:
The problem is that those options differ for editing and creating a room. I don't know if you refer to `room_properties` type which in this case is a simple struct (I guess it should be a map) or to `t()` which has lots of fields that are not available for editing and probably for creating as well. Moreover Janus VideoRoom api had to add `new` prefix in editing request which in this case makes it invalid to use same type (that is not a simple map) for both `create` and `edit`.